xdsl/dialects/hw.py

```diff
@@ -112,6 +121,142 @@ def print_parameters(self, printer: Printer) -> None:
         printer.print_string(">")
 
 
+class InnerSymbolTable(OpTrait):
+    """A trait for inner symbol table functionality on an operation.
+    Merges the upstream table of inner symbols and their resolutions and the op trait.
+    """
+
+    @overload
+    def __init__(self) -> None:
+        # Create an empty table, not used for lookups
+        ...
+
+    @overload
+    def __init__(self, op: Operation) -> None:
+        ...
```

Comment:
I'm not sure that these give any functional difference to the non-overloaded version. What changes if they are removed?